PROTO_tests/tests/series_test.py

```diff
@@ -0,0 +1,224 @@
+
+    def test_add(self):
+        i = ak.arange(3)
+        v = ak.arange(3, 6, 1)
+        s = ak.Series(data=i, index=i)
+
+        s_add = ak.Series(data=v, index=v)
+
+        added = s.add(s_add)
+
+        idx_list = added.index.to_pandas().tolist()
+        val_list = added.values.to_list()
+        for i in range(6):
+            assert i in idx_list and val_list
+
+    def test_topn(self):
+        v = ak.arange(3)
+        i = ak.arange(3)
+        s = ak.Series(data=v, index=i)
+
+        top = s.topn(2)
+        assert top.index.to_pandas().tolist() == [2, 1]
+        assert top.values.to_list() == [2, 1]
+
+    def test_sort_idx(self):
+        v = ak.arange(5)
+        i = ak.array([3, 1, 4, 0, 2])
+        s = ak.Series(data=v, index=i)
+
+        sorted = s.sort_index()
+        assert sorted.index.to_pandas().tolist() == [i for i in range(5)]
+        assert sorted.values.to_list() == [3, 1, 4, 0, 2]
+
+    def test_sort_value(self):
+        v = ak.array([3, 1, 4, 0, 2])
+        i = ak.arange(5)
+        s = ak.Series(data=v, index=i)
+
+        sorted = s.sort_values()
+        assert sorted.values.to_list() == [i for i in range(5)]
+        assert sorted.index.to_pandas().tolist() == [3, 1, 4, 0, 2]
+
+    def test_head_tail(self):
+        v = ak.arange(5)
+        i = ak.arange(5)
+        s = ak.Series(data=v, index=i)
+
+        head = s.head(2)
+        assert head.index.to_pandas().tolist() == [0, 1]
+        assert head.values.to_list() == [0, 1]
+
+        tail = s.tail(3)
+        assert tail.index.to_pandas().tolist() == [2, 3, 4]
+        assert tail.values.to_list() == [2, 3, 4]
+
+    def test_value_counts(self):
+        v = ak.array([0, 0, 1, 2, 2])
+        i = ak.arange(5)
+        s = ak.Series(data=v, index=i)
+
+        c = s.value_counts()
+        assert c.index.to_pandas().tolist() == [0, 2, 1]
+        assert c.values.to_list() == [2, 2, 1]
+
+        c = s.value_counts(sort=True)
+        assert c.index.to_pandas().tolist() == [0, 2, 1]
+        assert c.values.to_list() == [2, 2, 1]
+
+    def test_concat(self):
+        v = ak.arange(5)
+        i = ak.arange(5)
+        s = ak.Series(data=v, index=i)
+
+        v = ak.arange(5, 11, 1)
+        i = ak.arange(5, 11, 1)
+        s2 = ak.Series(data=v, index=i)
+
+        c = ak.Series.concat([s, s2])
+        assert c.index.to_pandas().tolist() == [i for i in range(11)]
+        assert c.values.to_list() == [i for i in range(11)]
+
+        df = ak.Series.concat([s, s2], axis=1)
+        assert isinstance(df, ak.DataFrame)
+
+        ref_df = pd.DataFrame(
+            {
+                "idx": [i for i in range(11)],
+                "val_0": [0, 1, 2, 3, 4, 0, 0, 0, 0, 0, 0],
+                "val_1": [0, 0, 0, 0, 0, 5, 6, 7, 8, 9, 10],
+            }
+        )
+        assert ((ref_df == df.to_pandas()).all()).all()
+
+    def test_pdconcat(self):
+        v = ak.arange(5)
+        i = ak.arange(5)
+        s = ak.Series(data=v, index=i)
```

Comment:
I notice each pair of (v, i) vars in this function are the same.  Should we combine them instead of assigning the same values twice?

Reply:
Updated